### DISCUSSION TITLE
Automatic resizing of projected image

### DISCUSSION THREAD

**Discussion by joostlammers** [ImageMagick/ImageMagick#2767] [Help]
Hi, 

we've been using ImageMagick quite some time already and stumbled into a requirement we don't seem to be able to solve.

So we have a transparant shape (could be anything, a heart, a triangle, etc) and we want to scale a rectangular image into it, the best way it fits, without cutting its borders. 
So we don't want the triangle overlayed onto our picture but the other way around. 

We know the width and height of the original image and the shape, but if the original image's ratio can be different each time.
The final result is solely about the image. The shape is just to get the size right.

![Untitled Diagram](https://user-images.githubusercontent.com/20766413/97409098-637f1f00-18fd-11eb-9c1a-bcb4a652e51c.png)

**Comment by snibgo** [comment]
What version of IM, on what platform?

"-define trim:percent-background=0%" can be used to find the rectangle width, height and offsets.

**Comment by fmw42** [comment]
If on Linux-like system (Linux, Mac OSX or Windows 10 or Windows w/Cygwin) you can try my innercrop script at my web site http://www.fmwconcepts.com/imagemagick/index.php
